rasa/core/test.py

```diff
@@ -336,11 +337,13 @@ async def _create_data_generator(
 ) -> "TrainingDataGenerator":
     from rasa.shared.core.generator import TrainingDataGenerator
 
-    from rasa.core import training
-
-    story_graph = await training.extract_story_graph(
+    test_data_importer = TrainingDataImporter.load_from_dict(
+        training_data_paths=[resource_name]
+    )
+    story_graph = await test_data_importer.get_stories(
         resource_name, agent.domain, use_e2e
```

Comment:
the parameter usage doesn't look right as PyCharm gives a warning.

Reply:
Ok, I will check!

Reply:
Thank you for pointed out the parameters usage issue. I changed the arguments and left only the `use_e2e` in the `get_stories` call but I still got the same error. 
After checking again the error message, I also saw that when running the `test_source_in_failed_stories` the assert  is related to the path itself. It looks like the difference is that I have the absolute path and the assert compares only with the root path. So I did a small change in the path [here](https://github.com/RasaHQ/rasa/blob/88d97e1d54ce79b9dac2d1091c7db7eb80fc6451/rasa/shared/data.py#L150).
As a note, the old way of loading the data was working because it does not use this function when running this test.  
After doing the change I run all the tests locally and it looks good. In the CI/CD I think the error is not related to the change but I will check again.

Reply:
> As a sidenote, pathlib (new standard library in Python 3.4) looks like a good alternative to the os.path module.

Yep, that's what we are trying to use in any new code 👍🏻 

Unfortunately it's very hard for me to reconstruct your issue due to the wrong diff of the PR 😬

Reply:
Thanks for the detailed explanation 💯